crypto/asn1/a_utf8.c

```diff
@@ -73,40 +73,6 @@ int UTF8_getc(const unsigned char *str, int len, unsigned long *val)
         if (value < 0x10000)
             return -4;
         ret = 4;
-    } else if ((*p & 0xfc) == 0xf8) {
```

Comment:
Do you have any idea of what the 5 and 6 byte cases were used for ??
This file is 20+ years old with no comment as to what those were :(

Reply:
I wouldn't be surprised if this is a case of "because we can". Logically speaking, the way utf8 works, it does along up to 6 bytes when encoding. That's a larger range the current Unicode range.

Reply:
With the proliferation of emoji creation, I'm not sure that 48 bits will be sufficient for long :grin:

Reply:
[This link](https://en.wikipedia.org/wiki/UTF-8#History) is enlightening: Originally, UTF-8 encoded all ode points from 0 to 0x7FFFFFFF (which fit in 6 bytes).

> In November 2003, UTF-8 was restricted by RFC 3629 to match the constraints of the UTF-16 character encoding: explicitly prohibiting code points corresponding to the high and low surrogate characters removed more than 3% of the three-byte sequences, and ending at U+10FFFF removed more than 48% of the four-byte sequences and all five- and six-byte sequences.

Reply:
That said, should I also include a test for the surrogate range and other invalid bytes?

Reply:
I think it would be useful.

Reply:
See the current branch.